bio2zarr/vcz.py

```diff
@@ -240,16 +219,24 @@ class VcfZarrSchema(core.JsonDataclass):
     samples_chunk_size: int
     variants_chunk_size: int
     fields: list
+    defaults: dict
 
     def __init__(
         self,
         format_version: str,
         fields: list,
         variants_chunk_size: int = None,
         samples_chunk_size: int = None,
+        defaults: dict = None,
     ):
         self.format_version = format_version
         self.fields = fields
+        defaults = defaults if defaults is not None else {}
+        if "compressor" not in defaults or defaults["compressor"] is None:
+            defaults["compressor"] = DEFAULT_ZARR_COMPRESSOR.get_config()
```

Comment:
We're overwriting the input ``defaults`` argument here which is a potential footgun. Easiest approach is to take a copy above.

Reply:
Nice catch - fixed